mail/src/main/java/jakarta/mail/internet/UniqueValue.java

```diff
@@ -47,12 +48,19 @@
      */
     public static String getUniqueBoundaryValue() {
 	StringBuilder s = new StringBuilder();
-	long hash = s.hashCode();
-
+        s.append("----=_Part_");
+        String p = PropUtil.getSystemProperty(
+                "mail.mime.multipart.boundary.factory", null);
+        if (UniqueValue.class.getName().equals(p)) {
```

Comment:
if `UUID` is the default, then it would be good to make it first in the `if`s chain (one may call it premature optimization... :-) )